src/main/java/com/backend/webproject/managers/PaymentManager.java

```diff
@@ -1,31 +1,45 @@
 package com.backend.webproject.managers;
 
+import javax.servlet.http.HttpServletRequest;
+
 import com.backend.webproject.dao.PaymentDAO;
 import com.backend.webproject.entity.Payment;
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.PathVariable;
 
 @Component
 public class PaymentManager {
-    @Autowired
-    PaymentDAO dao;
-
-    public int processPayment(int openCoupon, int shoppingFinalAmount, String paymentStatus, int shoppingCartID){
-        Payment payment = new Payment();
-        payment.setOpenCoupon(openCoupon);
-        payment.setShoppingFinalAmount(shoppingFinalAmount);
-        payment.setPaymentStatus(paymentStatus);
-        payment.setShoppingCartID(shoppingCartID);
-        return dao.processPayment(payment);
-    }
-
-    public Payment searchPaymentByID(int pid){
-        return dao.searchPaymentByID(pid);
-    }
-
-    public Payment searchPaymentByCartID(int cartID){
-        return dao.searchPaymentByCartID(cartID);
-    }
-    
-}
+
+	@Autowired
+	private PaymentDAO paymentJDBC;
+
+	@Autowired
+	private Payment payment;
+
+	public Payment searchByID(@PathVariable("id") int id) {
+		Payment pay = paymentJDBC.searchPaymentByID(id);
+
+		return pay;
+	}
+
+	public Payment searchByCartID(@PathVariable("id") int id) {
+		Payment pay = paymentJDBC.searchPaymentByCartID(id);
+
+		return pay;
+	}
+
+	public int processPayment(HttpServletRequest req) {
+		// int openCoupon = Integer.parseInt(req.getParameter("openCoupon"));
+		// int shoppingAmount = Integer.parseInt(req.getParameter("shoppingAmount"));
+		// String paymentStatus = req.getParameter("paymentStatus");
+		// int shoppingCartID = Integer.parseInt(req.getParameter("shoppingCartID"));
+
+		if (paymentJDBC.processPayment(payment) == 1) {
+			// Create shopping cart
+			return 1;
+		}
+		return 0;
```

Comment:
You can return directly the function call;

Reply:
Ohh wait, just saw the comment about creating the shopping cart. In that case you can leave it like this, just consider it.